### ISSUE TITLE
`default_gui_startup_args` has no effect when launching with desktop file

### ISSUE THREAD

**Issue by ADIX7** [wez/wezterm#2933]
### What Operating System(s) are you seeing this problem on?

Linux X11

### Which Wayland compositor or X11 Window manager(s) are you using?

Gnome

### WezTerm version

wezterm 20221119-145034-49b9839f

### Did you try the latest nightly build to see if the issue is better (or worse!) than your current version?

Yes, and I updated the version box above to show the version of the nightly that I tried

### Describe the bug

Using the desktop file, wezterm does not respects the `default_gui_startup_args` options. Starting from a run window or from another terminal with only `wezterm` command it does respect. If I remove the `start --cwd .` part from the desktop file's Exec line it does work.

### Configuration

Relevant part:

```
local wezterm = require 'wezterm';

return {
    unix_domains = { {
        name = 'unix'
    } },
    
    default_gui_startup_args = { 'connect', 'unix' }
}
```

**Comment by wez** [comment]
regarding my comment about the changes in `main`:

It typically takes about an hour before commits are available as nightly builds for all platforms.  Linux builds are the fastest to build and are often available within about 20 minutes.  Windows and macOS builds take a bit longer.

Please take a few moments to try out the fix and let me know how that works out. You can find the nightly downloads for your system in the [wezterm installation docs](https://wezfurlong.org/wezterm/installation.html).

If you prefer to use packages provided by your distribution or package manager of choice and don't want to replace that with a nightly download, keep in mind that you can download portable packages (eg: a `.dmg` file on macOS, a `.zip` file on Windows and an `.AppImage` file on Linux) that can be run without permanently installing or replacing an existing package, and can then simply be deleted once you no longer need them. 

If you are eager and can [build from source](https://wezfurlong.org/wezterm/install/source.html) then you may be able to try this out more quickly.

**Comment by ADIX7** [comment]
It does work with the latest nightly build (Appimage, tested on Fedora 37).

Can the multiplexer built into the gui process save and restore the state when closing the window like the unix socket?

**Comment by wez** [comment]
> Can the multiplexer built into the gui process save and restore the state when closing the window like the unix socket?

No, closing the window will (by default: after prompting you about stateful processes!) kill the associated local panes.

You can use the workspaces feature to switch to a different workspace, however, workspaces have to contain at least one window, so if you want banish all windows temporarily, this wouldn't be a 100% match to your desired workflow.

**Comment by Lenbok** [comment]
This method isn't working for me. Sometimes I get a window on the correct domain, and sometimes I get one on termwizterminaldomain. The windows that are opened also do not seem to obey the config `initial_rows` and `initial_cols`.
